app/src/common/key_and_signatures.c

```diff
@@ -174,14 +175,13 @@ static int ecdsa_sign_hash_and_zero_out_private_key(
       
         }
         CATCH_OTHER(e) { error = e; }
-        FINALLY { explicit_bzero(privateKey, sizeof(privateKey)); }
+        FINALLY { /* Do nothing, but make sure to zero out private key from calling function */  }
```

Comment:
This could be removed now that it does nothing, just retaining the comment.  Likely the `FINALLY` macro inserts some code here that could be avoided.

Reply:
Unfortunatley Ledger **requires** the existance of a `FINALLY`, the code does not compile without it. I get error:

```
.//src/common/key_and_signatures.c:194:3: error: function definition is not allowed here
) {
  ^
.//src/common/key_and_signatures.c:248:3: error: function definition is not allowed here
) {
  ^
.//src/common/key_and_signatures.c:283:63: error: function definition is not allowed here
                                         const uint8_t *hash) {
                                                              ^
.//src/common/key_and_signatures.c:310:2: error: expected '}'
}
 ^
.//src/common/key_and_signatures.c:150:3: note: to match this '{'
) {
  ^
.//src/common/key_and_signatures.c:176:9: error: use of undeclared label '__FINALLYEX'
        CATCH_OTHER(e) { error = e; }
        ^
/project/deps/nanos-secure-sdk/include/os.h:822:24: note: expanded from macro 'CATCH_OTHER'
#define CATCH_OTHER(e) CATCH_OTHER_L(EX, e)
                       ^
/project/deps/nanos-secure-sdk/include/os.h:556:8: note: expanded from macro 'CATCH_OTHER_L'
  goto CPP_CONCAT(__FINALLY, L);                                               \
       ^
/project/deps/nanos-secure-sdk/include/os.h:508:26: note: expanded from macro 'CPP_CONCAT'
#define CPP_CONCAT(x, y) CPP_CONCAT_x(x, y)
                         ^
/project/deps/nanos-secure-sdk/include/os.h:509:28: note: expanded from macro 'CPP_CONCAT_x'
#define CPP_CONCAT_x(x, y) x##y
                           ^
<scratch space>:76:1: note: expanded from here
__FINALLYEX
^
2 warnings and 5 errors generated.
/project/deps/nanos-secure-sdk/Makefile.rules_generic:64: recipe for target 'obj/key_and_signatures.o' failed
```

Reply:
And JUST by readding the (empty) `FINALLY {}`, the code compiles 🤷‍♂️

Reply:
As far as I can see from the headers, FINALLY is a required part since it closes whole TRY/CATCH/FINALLY sequence.

Reply:
Erf.  OK, ignore what I said then.